Core/Security/X509Request.py

```diff
@@ -4,33 +4,36 @@
 __RCSID__ = "$Id$"
 
 import GSI
+import M2Crypto
 from DIRAC import S_OK, S_ERROR
 from DIRAC.Core.Utilities import DErrno
 from DIRAC.Core.Security.X509Chain import X509Chain
 
 class X509Request( object ):
+  """
+  Class representing X509 Certificate Request
+  """
 
   def __init__( self, reqObj = None, pkeyObj = None ):
     self.__valid = False
     self.__reqObj = reqObj
     self.__pkeyObj = pkeyObj
-    if reqObj and pkeyObj:
+    if reqObj and pkeyObj:  # isn't it a bit too liberal?
       self.__valid = True
 
-  # It is not used
-  # def setParentCerts( self, certList ):
-  #   self.__cerList = certList
-
   def generateProxyRequest( self, bitStrength = 1024, limited = False ) :
-    self.__pkeyObj = GSI.crypto.PKey()
-    self.__pkeyObj.generate_key( GSI.crypto.TYPE_RSA, bitStrength )
-    self.__reqObj = GSI.crypto.X509Req()
+    """
+    Generate proxy request
+    """
+    self.__pkeyObj = M2Crypto.EVP.PKey()
+    self.__pkeyObj.assign_rsa(M2Crypto.RSA.gen_key(bitStrength, 65537, callback = M2Crypto.util.quiet_genparam_callback ))
+    self.__reqObj = M2Crypto.X509.Request()
     self.__reqObj.set_pubkey( self.__pkeyObj )
     if limited:
-      self.__reqObj.get_subject().insert_entry( "CN", "limited proxy" )
+      self.__reqObj.get_subject().add_entry_by_txt( field = "CN", type = M2Crypto.ASN1.MBSTRING_ASC, entry =  "limited proxy", len=-1, loc=-1, set=0 )
     else:
-      self.__reqObj.get_subject().insert_entry( "CN", "proxy" )
-    self.__reqObj.sign( self.__pkeyObj, "SHA256" )
+      self.__reqObj.get_subject().add_entry_by_txt( field = "CN", type = M2Crypto.ASN1.MBSTRING_ASC, entry =  "proxy", len=-1, loc=-1, set=0 )
+    self.__reqObj.sign( self.__pkeyObj, "sha256" )
```

Comment:
upper case (old) or lower case (new) are the same?

Reply:
M2Crypto doesn't recognise upper case name.